pkg/scheduling/requirements.go

```diff
@@ -97,6 +98,14 @@ func NewPodRequirements(pod *v1.Pod) Requirements {
 	return NewNodeSelectorRequirements(requirements...)
 }
 
+func NewAMIRequirements(ami v1alpha1.AMI) Requirements {
```

Comment:
We don't want AWS specific things leaking into the core scheduling logic. Seems like this is just a small helper, can you move this to the AWS cloudprovider?